geopandas/tools/tests/test_sjoin.py

```diff
@@ -75,7 +75,9 @@ def test_inner(self, op, dfs):
         index, df1, df2, expected = dfs
 
         res = sjoin(df1, df2, how='inner', op=op)
-
+        assert df1.index.name == None
+        assert df2.index.name == None
```

Comment:
I think testing that the index name is none in all the `sjoin` tests is probably overkill. The new test you added should suffice.